tests/test_registry.py

```diff
@@ -645,6 +645,58 @@ def raise_exception():
         assert "error" in response
         assert random_val == response["error"]["data"]["random"]
 
+    def test_encoder_exception_batch(self):
+        def custom_encode(input):
+            if "foo" in str(input):
+                raise InternalError("Could not parse the input data.")
+            else:
+                return json.JSONEncoder().encode(input)
+
+        registry = Registry()
+        registry.json_encoder.encode = custom_encode
+
+        fake_batch_request = self._create_fake_request([{
+            "jsonrpc": "2.0",
+            "method": "rpc.describe",
+            "params": [],
+            "id": "foo",
+        }, {
+            "jsonrpc": "2.0",
+            "method": "rpc.describe",
+            "params": [],
+            "id": "bar",
+        }])
+        response = json.loads(registry.dispatch(fake_batch_request))
+
+        assert isinstance(response, list)
+        assert len(response) == 2
+
+        foo_response = response[0] if response[1]["id"] == "bar" else response[1]
+        TestDispatch.assert_error(foo_response, "foo", InternalError)
+        assert foo_response["error"]["data"] == "Could not parse the input data."
+
+        bar_response = response[1] if response[1]["id"] == "bar" else response[0]
+        assert "error" not in bar_response
+
+    def test_encoder_exception_single(self):
+        def throw_error(input):
+            raise InternalError("Could not parse the input data.")
+
+        registry = Registry()
+        registry.json_encoder.encode = throw_error
```

Comment:
This test should ruin all of the other tests (unless there's some magic reloading). You should use `@pytest.patch` to replace the encode function for the scope of this test.